### PR TITLE
Restructuring and upgrading + move to npm theme/config!

### PR DESCRIPTION
**Pull request by abernix**
This PR aims to bring the Galaxy docs up to the same status as the Meteor Docs and Meteor Guide by not only becoming more consistent with their configuration, but by removing the need for future changes by moving to an [npm-powered theme](https://www.npmjs.com/package/meteor-theme-hexo) ([:octocat: repo](https://github.com/meteor/hexo-theme-meteor)) and inherited configuration (also powered by an [npm package](https://www.npmjs.com/package/meteor-hexo-config) ([:octocat: repo](https://github.com/meteor/meteor-hexo-config))) pattern which allows us to change global configurations (like links, colors, even theme) at one global location for all Meteor (and other MDG repositories).

I've also gone ahead and setup Netlify on this repository, so if all goes well, this PR will kick off a Deploy Preview, allowing us to confirm that the Netlify deployment is working properly.  If that deploy looks okay, and we merge this PR, I can flip the DNS for `galaxy-guide.meteor.com` over to the Netlify deployment (and off the S3 bucket), in the same way I have for Meteor Docs and Guide.  This should be the last repository needing this change and free us from using Amazon S3.  (Though, let's be honest, those deploy previews are _super_ cool for docs changes.)

I hope the verbosity of these commits does any additional explanation necessary.  Many of their commit messages have been borrowed from similar commits I've made (successfully!) on other repositories.

Ref: https://github.com/meteor/docs/pull/187
Ref: https://github.com/meteor/hexo-theme-meteor/pull/51

### CODE REVIEW

**Comment by abernix** [comment]
As seen above, the Netlify deploy preview seems to be demonstrating the success of this PR.  My confidence in these changes is quite high, though I'd, of course, always appreciate a review!

File: package.json [hunk 1]
```diff
@@ -12,10 +12,10 @@
     "hexo-generator-category": "^0.1.2",
     "hexo-generator-index": "^0.2.0",
     "hexo-generator-tag": "^0.2.0",
-    "hexo-renderer-ejs": "^0.2.0",
+    "hexo-renderer-ejs": "0.3.1",
```
Comment (glasser) [inline package.json]: Not ^ for these deps?

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "hexo-s3-deploy": "meteor/hexo-s3-deploy#e36f6d046660040878bb5bedfc8422f526a78be6"
   },
   "scripts": {
+    "test":"hexo clean; hexo generate",
```
Comment (glasser) [inline package.json]: Use `&&` rather than `;`?

**Comment by glasser** [comment]
(PS THIS IS OTHERWISE SUPER AWESOME, I HATE SUBMODULES AND RANDOM NONSTANDARD THINGS, YOU ARE THE BEST)

File: package.json (hunk 1 above)
Comment (abernix) [reply package.json]: Good eye, and good question.  This is to accommodate [Renovate](renovateapp.com) (bot), which I'll add to this repo soon.

File: package.json (hunk 2 above)
Comment (abernix) [reply package.json]: In retrospect, yes.  When I first started this adventure of updating all the repos, this command had been a lot more complicated, and it wasn't clear to me whether or not `clean` returned an error code for any meaningful, expected reason.  I don't think it does, and I think any failure at this point would probably be notable enough to switch this to `&&`.

I will follow-up with a ninja commit which makes that change across all the repos.

**Comment by abernix** [comment]
Rebased to avoid losing f2af699c30f3e2ea2615476ca63b9484b52868aa.

**Comment by abernix** [comment]
DNS has been switched.

**Pull request merged**